### PR TITLE
Trim column headers by default, and bump major version

### PR DESCRIPTION
**Pull request by jonathanmatthews**
Add the option to trim column headers when reading from the input text, and make it the default. This is a potentially breaking change, so increment the major version as well.

### CODE REVIEW

File: src/DelimitedDataParser/DelimitedDataReader.cs [hunk 1]
```diff
@@ -37,12 +39,14 @@ internal class DelimitedDataReader : DbDataReader
             Encoding encoding,
             char fieldSeparator,
             bool useFirstRowAsColumnHeaders,
+            bool trimColumnHeaders,
```
Comment (mdsharpe) [inline src/DelimitedDataParser/DelimitedDataReader.cs:42]: Worth making optional? Default to `true`?

Comment (mdsharpe) [reply src/DelimitedDataParser/DelimitedDataReader.cs:42]: I acknowledge that `useFirstRowAsColumnHeaders` seems equally default-able but has no default..

File: src/DelimitedDataParser/DelimitedDataReader.cs [hunk 2]
```diff
@@ -601,7 +605,14 @@ private void EnsureInitialised()
         private void GenerateFieldLookup()
         {
             // Here we assume that the current row is the header row.
-            _fieldNameLookup = new List<string>(_currentRow).AsReadOnly();
+            if (_trimColumnHeaders)
+            {
+                _fieldNameLookup = _currentRow.Select(o => o == null ? o : o.Trim()).ToList().AsReadOnly();
```
Comment (mdsharpe) [inline src/DelimitedDataParser/DelimitedDataReader.cs]: Could
```c#
Select(o => o == null ? o : o.Trim())
```
be shortened to
```c#
Select(o => o?.Trim())
```

File: src/DelimitedDataParser/DelimitedDataReader.cs (hunk 1 above)
Comment (mdsharpe) [reply src/DelimitedDataParser/DelimitedDataReader.cs:42]: I have now realized that this class is typically not constructed publicly so making the parameter default wouldn't really yield a lot of benefit.

**Review by mdsharpe — CHANGES_REQUESTED** [review]

File: src/DelimitedDataParser/DelimitedDataReader.cs (hunk 2 above)
Comment (jonathanmatthews) [reply src/DelimitedDataParser/DelimitedDataReader.cs]: Oops. Fixed.

**Review by mdsharpe — APPROVED** [review]

**Pull request merged**